pkg/cloudprovider/aws/aws.go

```diff
@@ -601,3 +614,42 @@ func addASGTags(config *cloudprovider.NodeGroupConfig, asg *autoscaling.Group, p
 		log.Errorf("failed to create auto scaling tag for ASG %v", id)
 	}
 }
+
+// terminateOrphanedInstances will attempt to terminate a list of instances
+func terminateOrphanedInstances(n *NodeGroup, instances []*string) {
+	if len(instances) == 0 {
+		return
+	}
+	log.WithField("asg", n.id).Infof("terminating %v instance(s) that could not be attached to the ASG", len(instances))
+
+	var instanceIds []string
+	for i := 0; i < len(instances); i += terminateBatchSize {
```

Comment:
There's an existing batch implementation here that doesn't require minInt
https://github.com/atlassian/escalator/blob/25b9823f614d772deb12bcb07ac1e0fe4aa9cd1e/pkg/cloudprovider/aws/aws.go#L421
Maybe you could use that, if it fits?

Reply:
Yeah I think it is easier to follow, happy to leave it as this one. Less likely for lead to gotchas like the other one that mutates the instance slice.